### PR TITLE
Feature/people

### PR DESCRIPTION
**Pull request by ozBenTik**

### CODE REVIEW

File: core/src/main/java/data/movies/datasource/localstore/CreditsStore.kt [hunk 1]
```diff
@@ -11,14 +11,14 @@ import javax.inject.Singleton
 @Singleton
 class CreditsStore @Inject constructor() {
     // Map<movieId, actors>
-    private val _movieActors = MutableSharedFlow<Map<Int, List<Actor>>>(replay = 1)
+    private val _movieActors = MutableSharedFlow<Map<Int, List<ActorPart>>>(replay = 1)
 
     @OptIn(ExperimentalCoroutinesApi::class)
```
Comment (Ronelg) [inline core/src/main/java/data/movies/datasource/localstore/CreditsStore.kt]: @OptIn(ExperimentalCoroutinesApi::class) is redundant

File: core/src/main/java/data/people/PeopleRepository.kt [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+import javax.inject.Inject
+
+class PeopleRepository @Inject constructor(
+    private val remote: PeopleRemoteDataSource,
+    private val local: PeopleLocalDataSource
+){
+
+    //------------ Popular capabilities ------------
+
+    fun observePopularPeople() = local.pPeopleStore.observeEntries()
```
Comment (Ronelg) [inline core/src/main/java/data/people/PeopleRepository.kt]: pPeopleStore is not good naming
should be camel case start with lower case letter

File: core/src/main/java/data/people/datasource/PeopleLocalDataSource.kt [hunk 3]
```diff
@@ -0,0 +1,8 @@
+package com.example.core.data.people.datasource
+
+import com.example.core.data.people.datasource.localstore.PPeopleStore
+import javax.inject.Inject
+
+class PeopleLocalDataSource @Inject constructor(
+    val pPeopleStore: PPeopleStore,
```
Comment (Ronelg) [inline core/src/main/java/data/people/datasource/PeopleLocalDataSource.kt]: rename pPeopleStore

File: core/src/main/java/data/people/datasource/PeopleRemoteDataSource.kt [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+import extensions.executeWithRetry
+import extensions.toResult
+import util.safeApiCall
+import javax.inject.Inject
+
+class PeopleRemoteDataSource @Inject constructor(
+    private val peopleService: PeopleService
+) {
+
+    suspend fun getPPeople(page: Int, language: String) =
```
Comment (Ronelg) [inline core/src/main/java/data/people/datasource/PeopleRemoteDataSource.kt]: getPPeople duplicate P

File: core/src/main/java/data/people/datasource/localstore/PPeopleStore.kt [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+import com.example.model.PopularActor
+import kotlinx.coroutines.ExperimentalCoroutinesApi
+import kotlinx.coroutines.flow.MutableSharedFlow
+import kotlinx.coroutines.flow.SharedFlow
+import kotlinx.coroutines.flow.asSharedFlow
+import javax.inject.Inject
+import javax.inject.Singleton
+
+@Singleton
+class PPeopleStore @Inject constructor() {
```
Comment (Ronelg) [inline core/src/main/java/data/people/datasource/localstore/PPeopleStore.kt]: PPeopleStore

File: core/src/main/java/di/StoreModule.kt [hunk 6]
```diff
@@ -45,4 +46,9 @@ class StoreModule {
     @Singleton
     @Provides
     fun provideActorsStore(): CreditsStore = CreditsStore()
+
+
+    @Singleton
+    @Provides
+    fun providePPeopleStore(): PPeopleStore = PPeopleStore()
```
Comment (Ronelg) [inline core/src/main/java/di/StoreModule.kt]: why do you need to provide this if you already create the class with inject const.?

File: core/src/main/java/network/PeopleService.kt [hunk 7]
```diff
@@ -0,0 +1,16 @@
...
+
+import com.example.model.PPeopleResponse
+import retrofit2.Call
+import retrofit2.http.GET
+import retrofit2.http.Query
+
+interface PeopleService {
+
+    @GET("person/popular")
+    fun getPPeople(
```
Comment (Ronelg) [inline core/src/main/java/network/PeopleService.kt]: rename getPPeople

File: domain/src/main/java/com/example/domain/people/PeoplePagingSource.kt [hunk 8]
```diff
@@ -0,0 +1,44 @@
...
+
+import androidx.paging.PagingSource
+import androidx.paging.PagingState
+import com.example.core.data.people.datasource.localstore.PPeopleStore
+import com.example.domain.movies.INITIAL_MOVIES_PAGE
+import com.example.model.PopularActor
+import timber.log.Timber
+import java.io.IOException
+
+class PPeoplePagingSource(
```
Comment (Ronelg) [inline domain/src/main/java/com/example/domain/people/PeoplePagingSource.kt]: rename PPeoplePagingSource

File: model/src/main/java/com/example/model/ActorPart.kt [hunk 9]
```diff
@@ -2,42 +2,42 @@ package com.example.model
...
-    val known_for_department: String,
+    val department: String? = null,
 
     @SerializedName("name")
-    val name: String,
+    val name: String? = null,
 
     @SerializedName("original_name")
-    val original_name: String,
+    val original_name: String? = null,
```
Comment (Ronelg) [inline model/src/main/java/com/example/model/ActorPart.kt]: rename variable name to camel case

File: model/src/main/java/com/example/model/ActorPart.kt [hunk 10]
```diff
@@ -2,42 +2,42 @@ package com.example.model
...
-    val popularity: Double,
+    val popularity: Double? = null,
 
     @SerializedName("profile_path")
-    val profile_path: String?,
+    val profilePath: String? = null,
 
     @SerializedName("cast_id")
-    val cast_id: Int,
+    val cast_id: Int? = null,
```
Comment (Ronelg) [inline model/src/main/java/com/example/model/ActorPart.kt]: rename variable name to camel case

File: model/src/main/java/com/example/model/ActorPart.kt [hunk 11]
```diff
@@ -2,42 +2,42 @@ package com.example.model
...
 
     @SerializedName("character")
-    val character: String,
+    val character: String? = null,
 
     @SerializedName("credit_id")
-    val credit_id: String,
+    val credit_id: String? = null,
```
Comment (Ronelg) [inline model/src/main/java/com/example/model/ActorPart.kt]: rename variable name to camel case

File: ui-movies/src/main/java/details/ActorsCarrouselAdapter.kt [hunk 12]
```diff
@@ -7,13 +7,13 @@ import androidx.recyclerview.widget.ListAdapter
 import androidx.recyclerview.widget.RecyclerView
 import com.bumptech.glide.Glide
 import com.example.core_ui.databinding.ActorCardBinding
-import com.example.model.Actor
+import com.example.model.ActorPart
 import com.example.model.util.TmdbImageUrlProvider
 
 class ActorsCarrouselAdapter(
```
Comment (Ronelg) [inline ui-movies/src/main/java/details/ActorsCarrouselAdapter.kt:13]: I asked not to do UI!!

File: ui-people/src/main/java/com/example/ui_people/PeopleLobbyViewModel.kt [hunk 13]
```diff
@@ -0,0 +1,59 @@
...
+import javax.inject.Inject
+
+@HiltViewModel
+class PeopleLobbyViewModel @Inject constructor(
+    private val pagingInteractor: ObservePagedPPeople,
+    private val logoutIteractor: LogoutIteractor,
+    private val dispatchers: AppCoroutineDispatchers
+) : ViewModel() {
+
+    private val language = MutableStateFlow("en-US")
```
Comment (Ronelg) [inline ui-people/src/main/java/com/example/ui_people/PeopleLobbyViewModel.kt:32]: hard coded string
You should create an object for your supported langs

**Review by Ronelg — CHANGES_REQUESTED** [review]

File: core/src/main/java/data/movies/datasource/localstore/CreditsStore.kt (hunk 1 above)
Comment (ozBenTik) [reply core/src/main/java/data/movies/datasource/localstore/CreditsStore.kt]: ok

File: core/src/main/java/data/people/datasource/PeopleLocalDataSource.kt (hunk 3 above)
Comment (ozBenTik) [reply core/src/main/java/data/people/datasource/PeopleLocalDataSource.kt]: Updated PPeople -> PopularPeople

**Pull request merged**